Share Site: Adjust copy based on feedback

danielbachhuber opened the pull request:
Fixes https://github.com/Automattic/dotcom-forge/issues/2593

## Proposed Changes

Adjusts the copy used in the Share Site for Preview components, and adds an inline support link:


### Sites Page Modal

<img width="649" alt="image" src="https://github.com/Automattic/wp-calypso/assets/36432/73e05c04-cea1-4c3d-bf2a-040b27e4c484">

<img width="634" alt="image" src="https://github.com/Automattic/wp-calypso/assets/36432/ed274795-d268-4ef7-9ffc-18b83c6fa04b">

### Privacy Settings, Launched

<img width="788" alt="image" src="https://github.com/Automattic/wp-calypso/assets/36432/58551db8-3605-4b1c-bf6c-aeee7f3c8c43">


<img width="781" alt="image" src="https://github.com/Automattic/wp-calypso/assets/36432/5794cacd-7b53-4661-9a9a-b607b3e5a285">

### Privacy Settings, Unlaunched

<img width="805" alt="image" src="https://github.com/Automattic/wp-calypso/assets/36432/c4b68139-961b-4685-bee3-8ea4876af942">


<img width="817" alt="image" src="https://github.com/Automattic/wp-calypso/assets/36432/0a493584-d1f9-400b-9267-7e4a8d0af96d">


## Testing Instructions

1. Create a new WordPress.com Business site.
2. Navigate to Sites, click on the ellipsis menu for your new site, click on 'Share site for preview'
3. Verify the modal appears and works as expected.
4. Navigate to General Settings for the site.
5. Verify the share site component appears as expected.

danielbachhuber committed:
Share Site: Adjust copy based on feedback

kozer commented:
Great work @danielbachhuber ! It works as expected.

One small note:
When the user is on Sites and the modal is open, when he clicks on "Learn more" a new modal opens. If the user closes the last modal, I would expect to show the previous one instead of having to go back to the ellipsis menu and reopen the modal.

It's not related to that ticket, so it's ok to skip that. If this is the case, it would be great if we create a new issue for that.

wojtekn reviewed (approved):
It's a helpful change! The code looks good and works as expected.

As we changed how the popup content is structured, it would make sense to change the loading state of the component:
- move the new text out of the loading indicator scope to show before we start loading data
- change Spinner to LoadingBar
We could do it as follow-up  PR, though.

In addition to @kozer note, I see one more annoying thing related to the 'Learn more' link - when I access it on the General Settings page, a window scrolls to the top.

javierarce commented:
Looking at the text again, I think we could shorten it a bit more: 


> 'Coming Soon' sites are only visible to you and invited users. Enable 'Share site' to let collaborators without an account view your site.

simison commented on client/components/site-preview-link/index.tsx:
```diff
@@ -106,17 +107,24 @@ export default function SitePreviewLink( {
 		<div>
+			<p>
+				{ translate(
+					"Sites in 'Coming Soon' mode are visible only to you and to the logged-in users you invite. Toggle 'Share site' to allow collaborators without an account to see your site. {{inlineSupportLink}}Learn more.{{/inlineSupportLink}}",
```
Should `'`s be `"`s instead?

simison replied:
"Coming Soon" in Title Case was also odd.

javierarce replied:
Good point, yes, we should use double quotes.

danielbachhuber replied:
Updated with cb2165420297108adb1337107aad092f16230901

danielbachhuber committed:
Merge branch 'trunk' into improve/share-site-link-copy

danielbachhuber committed:
Show two different descriptions based on context

danielbachhuber committed:
Use a `<LoadingPlaceholder />` instead of a `<Spinner />`

danielbachhuber commented:
> As we changed how the popup content is structured, it would make sense to change the loading state of the component:
> 
>     * move the new text out of the loading indicator scope to show before we start loading data
> 
>     * change Spinner to LoadingBar
>       We could do it as follow-up  PR, though.

Changed this behavior with 010fba495c1bfbe238bd7259e4b8ddc497f24676



> When the user is on Sites and the modal is open, when he clicks on "Learn more" a new modal opens. If the user closes the last modal, I would expect to show the previous one instead of having to go back to the ellipsis menu and reopen the modal.
>
> [...]
>
> In addition to @kozer note, I see one more annoying thing related to the 'Learn more' link - when I access it on the General Settings page, a window scrolls to the top.

Yeah, both of those are kind of annoying. I haven't figured out where they're coming from/how to fix yet.

@javierarce Do you think we should leave the "Learn more" in place or remove it?

javierarce commented:
> @javierarce Do you think we should leave the "Learn more" in place or remove it?

I think we could remove those links, the description and the text underneath the input field should be enough.

danielbachhuber committed:
Merge branch 'trunk' into improve/share-site-link-copy

danielbachhuber committed:
Drop the `<InlineSupportLink />`

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/8019144

Thank you @danielbachhuber for including a screenshot in the description! This is really helpful for our translators.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/8019144).
